tests/ui/UnreadIndicatorsTest.js

```diff
@@ -0,0 +1,395 @@
+
+beforeAll(() => {
+    global.fetch = TestHelper.getGlobalFetchMock();
+
+    // We need a bit more time for this test in some places
+    jest.setTimeout(30000);
+    Linking.setInitialURL('https://new.expensify.com/r/1');
+    appSetup();
+});
+
+/**
+ * @param {RenderAPI} renderedApp
+ */
+function scrollUpToRevealNewMessagesBadge(renderedApp) {
+    fireEvent.scroll(renderedApp.getByTestId('report-actions-list'), {
+        nativeEvent: {
+            contentOffset: {
+                y: 250,
+            },
+            contentSize: {
+                // Dimensions of the scrollable content
+                height: 500,
+                width: 100,
+            },
+            layoutMeasurement: {
+                // Dimensions of the device
+                height: 700,
+                width: 300,
+            },
+        },
+    });
+
+    // We advance the timer since we must wait for the animation to end
+    // and the new style to be reflected
+    jest.advanceTimersByTime(100);
+}
+
+/**
+ * @param {RenderAPI} renderedApp
+ * @return {Boolean}
+ */
+function isNewMessagesBadgeVisible(renderedApp) {
+    const badge = renderedApp.getByTestId('new-messages-badge');
+    return badge.props.style.transform[0].translateY === 10;
+}
+
+/**
+ * @param {RenderAPI} renderedApp
+ * @return {Promise}
+ */
+async function navigateToSidebar(renderedApp) {
+    const reportHeader = renderedApp.getByTestId('report-header');
+    const reportHeaderBackButton = reportHeader.findByType(ReportHeaderViewBackButton);
+    fireEvent(reportHeaderBackButton, 'press');
+    return waitForPromisesToResolve();
+}
+
+/**
+ * @param {RenderAPI} renderedApp
+ * @param {Number} index
+ * @return {Promise}
+ */
+async function navigateToSidebarOption(renderedApp, index) {
+    const optionRows = renderedApp.getAllByTestId('option-row');
+    fireEvent(optionRows[index], 'press');
+    await waitForPromisesToResolve();
+}
+
+/**
+ * @param {RenderAPI} renderedApp
+ * @return {Boolean}
+ */
+function isDrawerOpen(renderedApp) {
+    const reportScreen = renderedApp.getByTestId('report-screen');
+    return reportScreen.findByType(ReportActionsView).props.isDrawerOpen;
+}
+
+test('Unread LHN Status, “New Messages” badge, and New Line Indicator', async () => {
+    const REPORT_ID = 1;
+    const USER_A_ACCOUNT_ID = 1;
+    const USER_A_EMAIL = 'user_a@test.com';
+    const USER_B_ACCOUNT_ID = 2;
+    const USER_B_EMAIL = 'user_b@test.com';
+    const USER_C_ACCOUNT_ID = 3;
+    const USER_C_EMAIL = 'user_c@test.com';
+
+    // Render the App and sign in as a test user.
+    const renderedApp = render(<App />);
+
+    // Note: act() is necessary since react-navigation's NavigationContainer has an internal state update that will throw some
+    // warnings related to async code. See: https://callstack.github.io/react-native-testing-library/docs/understanding-act/#asynchronous-act
+    await act(async () => {
+        await waitForPromisesToResolve();
```

Comment:
Why are there two of these?

Reply:
🤦 because two is better than one? :trollface: (it's not needed)